### ISSUE TITLE
False positives for hard-coded secret rules

### ISSUE THREAD

**Issue by elsapet** [Bearer/bearer-rules#73]
## Description & Reproduction

We have three hard-coded secret rules for Ruby/Lang, JavaScript/Lang and JavaScript/Express (note: _not_ the built-in Gitleaks rule). These all match on patterns like `PASSWORD = <string_literal>`. This leads to false positives when we have the following kinds of things in our codebase: 

```
  export const CHANGE_PASSWORD = 'changePassword'

  googleMapsApiKey: GOOGLE_MAPS_KEY

  ERRORS = {
    WRONG_PASSWORD: "Your password is incorrect. Please try again."
  }
```

We should make a decision on how - if at all - we want to address these false positives. See Expected Behavior.

## Expected Behavior

To be decided. Some options: 

1. We accept that these kinds of false positives will exist for this rule, since the risk here is so great (exposed passwords). We leave it to the Bearer user to exclude these cases with a code comment or by excluding the relevant files or folders from their scan.
2. We try to exclude cases where, for example, the string literal includes "PASSWORD", "API_KEY", etc, or where the entropy of the string literal suggests it is not a secret.
3. We still catch these cases but with a default severity level of `warning`

## Actual Behavior

We catch these cases and raise findings with a default severity level of `high`. 

## Possible Fix

See Expected Behavior.

## Your Environment

- Operating System and version:
- Output of 'bearer version':
```
bearer version: 1.8.0
sha: df097795e3a435a9a0aa9127342d58fc20bf6416
```

**Comment by elsapet** [comment]
After some discussion, the preferred (ideal) solution to false positive cases like this would be for findings to include a "confidence level" of sorts, to indicate how sure we are that a certain finding is a true instance of a given security risk. 

This way, for example, the scanner could be set to ignore findings whose confidence levels fall below a certain threshold.
